scipy/linalg/basic.py

```diff
@@ -1125,3 +1125,146 @@ def pinvh(a, cond=None, rcond=None, lower=True, return_rank=False,
         return B, len(psigma_diag)
     else:
         return B
+
+def balance(A, permute=True, scale=True, separate = False,
+                 silent=False, overwrite_a=False, scaled_block=False):
+    """
+    A wrapper around LAPACK's xGEBAL routine family for matrix balancing. 
+    
+    The balancing tries to equalize the row and column 1-norms by applying 
+    a similarity transformation such that the magnitude variation of the 
+    matrix entries is reflected to the scaling matrices. This is particularly 
+    useful for eigenvalue and matrix decompositions and often automatically 
+    performed. Hence, this function provides a convenience for manually 
+    balancing. 
+        
+    Moreover, if turned on, the matrix is first permuted to isolate the upper 
+    triangular parts of the matrix and only the remaining part is scaled. 
+    
+    The balanced matrix satisfies the following equality 
+    
+    .. math::
+         
+                        B = T^{-1} A T 
+
+    The algorithm is based on the well-known technique of [1]_ and has 
+    been modified to account for special cases. See [2]_ for details
+    which have been implemented since LAPACK v3.5.0. 
+    
+    
+    Parameters
+    ----------
+    A : (n, n) array_like
+        Square data matrix for the balancing.
+        
+    permute : bool
+        The selector to define whether permutation of A is also performed
+        prior to scaling.
+    
+    scale : bool
+        The selector to turn on and off the scaling. If False, the matrix 
+        will not be scaled. 
+        
+    separate : bool
```

Comment:
That sounds better. Definitely agree with (1) and (3). I'm not quite sure about (2). Why not return both ``(scaling, perm)`` and ``np.diag(scaling)[perm,:]``? Also not super pretty, but removes another keyword, there's no extra computations and the memory taken up is easily discarded for those who care about that.